### PR TITLE
Update TimedLoop module

### PR DESCRIPTION
**Pull request by WareditorGit**
Added the possibility to have different periods between different module implementations by overriding the new getPeriod() function.

### CODE REVIEW

**Comment by Frotty** [comment]
I think the multiple timers kind of get rid of the performance benefit?
Maybe a timestamp could be preferred here?

**Comment by WareditorGit** [comment]
Maybe keep the original TimedLoop and add another module with different period per implementation possible?

One other idea would be to have timers associated to their period and when a class request a timer of a given period it would check if that timer already exist and run and would add the class function to the list of the callbacks of that timer. Might be overkill.

**Comment by WareditorGit** [comment]
I went with the second idea. Tell me what you think.

**Comment by Frotty** [comment]
Should be fine this way I suppose.
@Cokemonkey11 what do you think?

**Comment by Cokemonkey11** [comment]
I don't use this package, so:

- what's this for?
- what's the package for?
- where's the docs?
- why would you want to override the period?

**Comment by WareditorGit** [comment]
I didn't create the package, I just made it so the TimedLoop module could support different periods in different class implementation.

**Comment by Cokemonkey11** [comment]
The diff for that should be just a few lines, no?

- Not saying this can't be merged as-is, but
- I'm not the best person for that

File: wurst/util/TimerUtils.wurst [hunk 1]
```diff
@@ -45,75 +46,110 @@ public function timer.release()
...
+/* TimedLoop for classes */
 
-function evaluateTimedLoopTrig()
-	TriggerEvaluate(timedLoopTrig)
-	if conditionCount == 0
-		timedLoopTimer.pause()
+class TimedLoopTimer 
 
-enum Mode
+	constant timer loopTimer = getTimer()..setData(this castTo int)
```
Comment (Frotty) [inline wurst/util/TimerUtils.wurst]: should omit types here

**Comment by Frotty** [comment]
Thanks 👍

**Pull request merged**